core/transfer/src/test/java/org/eclipse/dataspaceconnector/transfer/core/provision/ProvisionManagerImplTest.java

```diff
@@ -14,57 +18,118 @@
 
 import java.util.List;
 
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.isA;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 class ProvisionManagerImplTest {
```

Comment:
we might wanna add a few tests, feel free to use this: https://gist.github.com/44cf444fda5c705823530fff6af5b779

Reply:
In async components testing the case where a future fails and then everything fails is kinda of implicit (at least, in this case).
However I added the two tests and improved assertions and names